--- main.mjs	2024-10-24 00:32:36.050599300 +0900
+++ main-pack.mjs	2024-10-24 21:32:41.146024800 +0900
@@ -9,6 +9,7 @@ import { CronJob } from 'cron'
 
 import { setting, setSetting, getSettingCmd, consoleLog, consoleWarn } from './setting.mjs'
 import { num37ToStr, getSmallMemoryFile, getZeroFill, aton6Start, aton4 } from './utils.mjs'
+import { update as updataDbAsync } from './db.mjs'
 
 const v4db = setting.v4
 const v6db = setting.v6
@@ -295,6 +296,21 @@ export const updateDb = (_setting, noRel
 		arg = getSettingCmd()
 	}
 
+	if(!_setting && !sync && !setting.smallMemory){
+		var sameDbSetting = setting.sameDbSetting
+		if(!sameDbSetting) setting.sameDbSetting = true
+		return updataDbAsync().then((r) => {
+			setting.sameDbSetting = sameDbSetting
+			if(r === true){
+				if(!noReload) reload()
+			}
+			return true
+		}).catch(e => {
+			setting.sameDbSetting = sameDbSetting
+			throw e
+		})
+	}
+
 	var scriptPath = path.resolve(_setting ? _setting.apiDir : setting.apiDir, 'script', 'updatedb.mjs')
 	if(scriptPath.includes(' ')) scriptPath = '"' + scriptPath + '"'
 	var cmd = 'node ' + scriptPath
